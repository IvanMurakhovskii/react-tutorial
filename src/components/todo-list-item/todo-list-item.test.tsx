import React from 'react';
import { shallow } from "enzyme";
import TodoListItem, { TodoListItemProps } from './todo-list-item';

const onDeletedMock = jest.fn();
const onToggleDoneMock = jest.fn();
const onToggleImportantMock = jest.fn();

const props: TodoListItemProps = {
    label: "todo",
    important: true,
    done: false,
    onDeleted: onDeletedMock,
    onToggleDone: onToggleDoneMock,
    onToggleImportant: onToggleImportantMock
}

const item = <TodoListItem {...props} />

describe("TodoListItem", () => {
    const wrapper = shallow(item);

    it("render item with data", () => {
        expect(wrapper).toMatchSnapshot();
    });

    it("simulate click impotrant button", () => {
        wrapper.find('.btn-impotrant').simulate('click');
        expect(onToggleImportantMock).toHaveBeenCalled();
    });

    it("simulate click delete button", () => {
        wrapper.find('.btn-delete').simulate('click');
        expect(onDeletedMock).toHaveBeenCalled();
    });

});
