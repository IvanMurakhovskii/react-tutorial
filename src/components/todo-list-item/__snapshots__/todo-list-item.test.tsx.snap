// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoListItem render item with data 1`] = `
<Styled(div)
  className="todo-list-item"
>
  <Styled(div)
    done={false}
    important={true}
    label="todo"
    onDeleted={[MockFunction]}
    onToggleDone={[MockFunction]}
    onToggleImportant={[MockFunction]}
  >
    <span
      onClick={[MockFunction]}
    >
      todo
    </span>
  </Styled(div)>
  <Styled(div)>
    <WithStyles(ForwardRef(Button))
      className="btn-delete"
      color="secondary"
      onClick={[MockFunction]}
      size="small"
      variant="contained"
    >
      Delete
    </WithStyles(ForwardRef(Button))>
    <WithStyles(ForwardRef(Button))
      className="btn-impotrant"
      color="primary"
      onClick={[MockFunction]}
      size="small"
      variant="contained"
    >
      Important
    </WithStyles(ForwardRef(Button))>
  </Styled(div)>
  <Styled(div)>
    <TodoTimer
      seconds={1500}
    />
  </Styled(div)>
</Styled(div)>
`;
